src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -334,8 +350,29 @@ public int onErrorHandling() {
       return 1;
     }
 
+    /**
+     * Gets the custom template specified by user which can be either a templateId or a full custom template path
+     *
+     * @return String of templateId or custom template path
+     */
+    public String getCustomTemplate() {
+      return Strings.isNullOrEmpty(templateId) ? customTemplatePath :
+        String.format("projects/%s/inspectTemplates/%s", getProject(), templateId);
+    }
 
     public void validate(FailureCollector collector, Schema inputSchema) {
+      if (customTemplateEnabled) {
+        if (!containsMacro(TEMPLATE_ID_NAME) && !containsMacro(CUSTOM_TEMPLATE_PATH_NAME) &&
+          Strings.isNullOrEmpty(templateId) && Strings.isNullOrEmpty(customTemplatePath)) {
+          collector.addFailure("Must specify one of the custom template fields", "")
+            .withConfigProperty(TEMPLATE_ID_NAME).withConfigProperty(CUSTOM_TEMPLATE_PATH_NAME);
+        }
+        if (!containsMacro(TEMPLATE_ID_NAME) && !containsMacro(CUSTOM_TEMPLATE_PATH_NAME) &&
+          !Strings.isNullOrEmpty(templateId) && !Strings.isNullOrEmpty(customTemplatePath)) {
+          collector.addFailure("Must specify only one of the custom template fields", "")
```

Comment:
collector.addFailure("Both template id and template path are specified", "Must specify only one of template id or template path")

Reply:
PR updated accordingly.